File: selectionsort.java
package assignments.selectionsort;
import java.util.Scanner;

public class selectionsort {

    void selection_sort(int array[]) {
        for (int i = 0; i < 5 - 1; i++) {
            int min = i;
            for (int j = i + 1; j < 5; j++) {
                if (array[j] < array[min]) {
                    min = j;
                }
            }
            int temp = array[i];
            array[i] = array[min];
            array[min] = temp;
        }
    }

    void show(int array[]) {
        System.out.println("Sorted array:");
        
        for (int i = 0; i < 5; i++) {
            System.out.print(array[i] + " ");
        }
    }

    public static void main(String[] args) {
        int arr[] = new int[5];
        Scanner sc = new Scanner(System.in);
        System.out.println("Enter an array:");
        for (int i = 0; i < 5; i++) {
            arr[i] = sc.nextInt();
        }
        selectionsort obj = new selectionsort();
        obj.selection_sort(arr);
        obj.show(arr);
    }
}

File: selectionsort.js
class SelectionSort {
    selectionSort(array) {
        for (let i = 0; i < array.length - 1; i++) {
            let min = i;
            for (let j = i + 1; j < array.length; j++) {
                if (array[j] < array[min]) {
                    min = j;
                }
            }
            let temp = array[i];
            array[i] = array[min];
            array[min] = temp;
        }
    }

    show(array) {
        console.log("Sorted array:");
        console.log(array.join(" "));
    }
}

const obj = new SelectionSort();
const arr = new Array(5); 

const readline = require('readline');
const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout
});

console.log("Enter an array:");

rl.question('', (input) => 
{
    arr = input.split(' ').map(Number);
    obj.selectionSort(arr);
    obj.show(arr);
    rl.close();
});
